extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java

```diff
@@ -331,14 +326,19 @@ private void configureCodecRegistry(CodecRegistry defaultCodecRegistry, MongoCli
         // it always needs to be the last codec provided
         PojoCodecProvider.Builder pojoCodecProviderBuilder = PojoCodecProvider.builder()
                 .automatic(true)
-                .conventions(Conventions.DEFAULT_CONVENTIONS);
+                // Note ANNOTATION_CONVENTION is the convention I found that works for my use case of _id & id in Mongo record.
+                // However, I think the fix should be that Quarkus allow users to set any number of conventions per connection
+                // as everyone's use case may be different.
+                .conventions(Arrays.asList(ANNOTATION_CONVENTION));
```

Comment:
This tree failed to build so I haven't run the tests but I do have some thoughts.  :)  Apart from some errant formatting changes (which maybe should be reverted but I don't feel strongly enough to fight for that), I think this is OK.  I would add back in `OBJECT_ID_GENERATORS` that comes from the defaults because that's a pretty big piece of behind the scenes work that would trip users up unexpectedly.